src/webglimpse/label.ts

```diff
@@ -148,10 +148,20 @@ export function newLabelPainter(label: Label, xFrac: number, yFrac: number, xAnc
             gl.clear(GL.COLOR_BUFFER_BIT);
         }
 
-        const texture = label.texture;
-        if (texture) {
+        if (truncateLabel) {
+            const xRight = viewport.w - 2 / viewport.w;
+            while (!!label.text && label.text !== '...') {
```

Comment:
Repeatedly destroying and creating textures in a while loop seems sub-optimal. Maybe we can just use the canvas `context.measureText()` function instead?

Not sure the `!!` before `label.text` is necessary.